docs/blog/2021-01-25-zmk-sotf-4.md

```diff
@@ -0,0 +1,195 @@
+:::
+
+#### Sticky Keys (One-Shot Mods/Layers) Behavior
+
+[okke-formsma] also contributed the initial "sticky keys" behavior, which can be used for functionality sometimes called "one shot mods" or "one shot layers". In your keymap, this would like like:
+
+```
+&sk LEFT_CONTROL
+```
+
+for a sticky key/modifier, or:
+
+```
+&sl NAV
+```
+
+for a sticky layer.
+
+#### `&to` Layer Behavior
+
+[mcrosson] contributed the new [`&to`](/docs/behaviors/layers#to-layer) layer related behavior. This can be used to completely replace the active layer with a new one.
+
+This is most frequently used when using multiple core base layers with different layouts, e.g. QWERTY and DVORAK, to switch between them.
+
+#### Grace Escape Behavior
+
+[okke-formsma] added an implementation of the "Grave Escape" behavior, developing a more generic "mod-morph" behavior to do so. Adding
+
+```
+&gresc
+```
+
+to your keymap will send `ESC` when pressed on its own, but will send `` ` `` when pressed with a GUI or Shift modifier held.
+
+#### Underglow Color Selection
+
+https://github.com/zmkfirmware/zmk/pull/514 [mcrosson]
+
+#### Keymap Upgrader
+
+[joelspadin] completed the [Keymap Upgrader](/docs/codes/keymap-upgrader) which can be used to update your keymap to using the latest supported codes, and move away from the old deprecated codes.
+
+If you've made keymap customizations, please make sure to run your keymaps through the upgrader, since the old deprecated codes will be removed in a future version of ZMK.
+
+### Displays
+
+There has been lots of work to get display support complete enough for use by end users. Although not quite ready for prime time, it is incredibly close, and we are looking forward to having the last few items completed and the feature documented!
+
+#### Idle Blanking
+
+[petejohanson] added idle blanking for displays, which ensures they will go blank, and into low power mode, after a short period of inactivity from the user. This ensures we avoid burn-in for OLEDs, and helps improve battery life.
+
+#### Battery and Output Widgets
+
+[petejohanson] implemented the first two complete, dynamic "widgets" for the displays for ZMK, adding a small battery indicator, which includes charging status, and a small output indicator, showing the currently active output (USB or BLE). When using BLE, the indicator also shows the active profile slot, as well as if the profile slot is open, awaiting connection from the paired host, or is actively connected to the host for that profile slot.
+
+#### Highest Layer Display
+
+[mcrosson] has contributed the next display widget, showing the highest active layer in the keymap. [petejohanson] then added a small follow up to allow layers in keymaps to add a `label` property to each layer, e.g. `label = "Nav";` and have that label be displayed in the widget instead of the numeric layer number.
```

Comment:
Are there any keymaps in the repo that demonstrate the use of layers with labels?

Reply:
I don't believe so but it's very straight forward. See [here](https://git.kemonine.info/kemonine/keyboard/src/branch/master/zmk-config/config/nibble.keymap) if you'd like an example.

With the OLED not enabled by default I'm not sure it's necessary to say more than what's already here.

It'll be good to include in the OLED docs when they are written but for now there aren't any docs on OLED usage.

Reply:
Not yet. Documenting all of this in details is definitely part of the work remaining to finish this.